functorch/csrc/BatchRulesConv.cpp

```diff
@@ -0,0 +1,83 @@
+#include <functorch/csrc/BatchRulesHelper.h>
+
+namespace at { namespace functorch {
+
+// batching rules translated from jax: https://github.com/google/jax/blob/master/jax/_src/lax/lax.py#L3143
+Tensor reshape_axis_into(int64_t src, int64_t dst, const Tensor &x) {
+  VmapDimVector new_shape(x.sizes().begin(), x.sizes().end());
+  new_shape.erase(new_shape.begin() + src);
+  new_shape[dst] *= x.sizes()[src];
+  return at::reshape(x.movedim(dst, src), new_shape);
+}
+
+Tensor reshape_axis_outof(int64_t src, int64_t size1, const Tensor &x) {
+  VmapDimVector shape(x.sizes().begin(), x.sizes().end());
+  TORCH_INTERNAL_ASSERT(shape[src] % size1 == 0);
+  int64_t size2 = shape[src] / size1;
+  shape[src] = size1;
+  shape.insert(shape.begin() + src + 1, size2);
+  return at::reshape(x, shape);
+}
```

Comment:
Btw, I needed these for something else so I added them in https://github.com/zou3519/functorch/commit/adc32e01d459bcb22ab2a782be112b4e67645f21 (and made it so that they can accept negative dims)